### PR TITLE
feat!: DBTP-1002 Remove support for 100% AWS Copilot version of DBT Platform

### PR DESCRIPTION
**Pull request by WillGibson**
Addresses [DBTP-1002](https://uktrade.atlassian.net/browse/DBTP-1002)

This is a huge breaking change. I would really appreciate a multi person review!

Huge documentation cleanup to follow.

Passing end to end tests run: https://eu-west-2.console.aws.amazon.com/codesuite/codepipeline/pipelines/willg-end-to-end-tests/executions/8104488d-f2f8-4502-9f2d-4948e05cf239/visualization?region=eu-west-2

---
### Title:
- [x] Scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Ticket reference included (unless it's a quick out of ticket thing)
### Description:
- [x] Link to ticket included (unless it's a quick out of ticket thing)
- [x] Includes tests (or an explanation for why it doesn't)
- ~[ ] If the work includes user interface changes, before and after screenshots included in description~
- ~[ ] Includes any applicable changes to the documentation in this code base~
- ~[ ] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)~
### Tasks:
- [x] [Trigger the pull request regression tests for this branch](https://github.com/uktrade/platform-tools?tab=readme-ov-file#regression-tests) and confirm that they are passing

[DBTP-1002]: https://uktrade.atlassian.net/browse/DBTP-1002?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dbt_platform_helper/utils/cloudformation.py [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
+#         command.extend(["--file", file])
+#
+#     if ignore_path:
+#         for ignored_file in glob.glob(ignore_path):
+#             command.extend(["--skip-path", ignore_path])
+#
+#     click.secho(f"\n>>> Running security check", fg="yellow")
+#     click.secho(f"""    {" ".join(command)}\n""", fg="yellow")
+#
+#     return run(command, capture_output=True)
```
Comment (james-francis-MT) [inline dbt_platform_helper/utils/cloudformation.py:34]: Will we need this commented out code at some point? Is there a reason not to delete?

File: tests/platform_helper/test_command_version.py [hunk 2]
```diff
@@ -1,76 +1,77 @@
...
-    def test_fails_if_pipeline_option_is_not_a_pipeline_given_invalid_config(
-        self, mock_latest_release
-    ):
-        command = VersionCommand().command
-        result = CliRunner().invoke(command, ["--pipeline", "bogus"])
-
-        assert result.exit_code != 0
-        assert "'bogus' is not " in result.output
-        assert "'prod-main'" in result.output
+# Todo: Uncomment these. Had to comment them out because they won't play in in my IDE :-(
```
Comment (james-francis-MT) [inline tests/platform_helper/test_command_version.py:1]: TODO for the PR or later

**Review by james-francis-MT — APPROVED** [review]

File: tests/platform_helper/test_command_version.py [hunk 3]
```diff
@@ -1,76 +1,77 @@
...
+#
+#     def test_fails_if_pipeline_option_is_not_a_pipeline_given_invalid_config(
+#         self, mock_latest_release
+#     ):
+#         command = VersionCommand().command
+#         result = CliRunner().invoke(command, ["--pipeline", "bogus"])
+#
+#         assert result.exit_code != 0
+#         assert "'bogus' is not " in result.output
+#         assert "'prod-main'" in result.output
```
Comment (antroy-madetech) [inline tests/platform_helper/test_command_version.py:77]: Need uncommenting?

File: dbt_platform_helper/utils/cloudformation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (WillGibson) [reply dbt_platform_helper/utils/cloudformation.py:34]: Yes, I forgot. Doing now.

File: tests/platform_helper/test_command_version.py (hunk 3 above)
Comment (WillGibson) [reply tests/platform_helper/test_command_version.py]: Done

File: dbt_platform_helper/templates/addon-instructions.txt [hunk 4]
```diff
@@ -5,7 +5,7 @@ Secret references:
   REDIS_ENDPOINT: /copilot/${COPILOT_APPLICATION_NAME}/${COPILOT_ENVIRONMENT_NAME}/secrets/{{ service.secret_name }}
 {%- elif service.addon_type == "opensearch" %}
   OPENSEARCH_CREDENTIALS: /copilot/${COPILOT_APPLICATION_NAME}/${COPILOT_ENVIRONMENT_NAME}/secrets/{{ service.secret_name }}
-{%- elif service.addon_type in ["aurora-postgres", "postgres"] %}
+{%- elif service.addon_type in ["postgres"] %}
```
[the remark this answers is not in the document]
Comment (WillGibson) [reply dbt_platform_helper/templates/addon-instructions.txt]: Done

File: dbt_platform_helper/utils/cloudformation.py (hunk 1 above)
Comment (WillGibson) [reply dbt_platform_helper/utils/cloudformation.py]: Deleted

**Review by james-francis-MT — APPROVED** [review]

**Comment by james-francis-MT** [comment]
> ### ❌ 48 Tests Failed:
> Tests completed	Failed	Passed	Skipped
> 606	48	558	0
> View the top 3 failed tests by shortest run time
> To view more test analytics, go to the [Test Analytics Dashboard](https://app.codecov.io/gh/uktrade/platform-tools/tests/DBTP-1002-remove-aws-copilot-cruft) Got feedback? Let us know on [Github](https://github.com/codecov/feedback/issues)

Does this thing update as you fix / remove / add tests in the PR?

**Review by antroy-madetech — APPROVED** [review]

**Comment by WillGibson** [comment]
> Does this thing update as you fix / remove / add tests in the PR?

Time will tell.

**Pull request merged**